### ISSUE TITLE
[Bug] MathJax 跨页面渲染

### ISSUE THREAD

**Issue by liblaf** [vuepress-theme-hope/vuepress-theme-hope#2790]
### Checklist

- [X] I am using **LTS** version of Node.js.
- [X] I am using the **latest v2 version** of VuePress and VuePress official plugins.
- [X] I am using the **latest v2 version** of vuepress-theme-hope and all plugins in this repo.
- [X] I am following the docs and I am sure I am using the correct configuration.
- [X] I have searched [FAQ](https://theme-hope.vuejs.press/faq/)

### Package name

vuepress-plugin-md-enhance

### Which operating system are you using?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Describe the bug

MathJax 似乎一次性渲染了不同页面的公式, 这导致同一页面内的公式编号不会从 1 开始. 如果不同页面使用了相同的 label, 比如 `\label{eq:1}` 出现在两个不同页面上, MathJax 就会报错.
理想的行为是: 对于每一个页面 MathJax 都会被初始化, 页面间的公式渲染是独立的.

**Comment by Mister-Hope** [comment]
你的理想行为会导致 N 倍的样式输出。关于细节可以详见 https://mdit-plugins.mrhope.site/zh/mathjax.html，这里的设计是让样式在页面之间复用并避免重复输出。

不过基于你的汇报，这似乎会产生标签问题？

这样的话，我也没有什么好的修复意见，除非将mathjax移动至客户端进行动态渲染。

**Comment by Mister-Hope** [comment]
cc @init-qy 

I found a related issue on https://github.com/mathjax/MathJax/issues/2931 and it mentions api which clears label.

**Comment by liblaf** [comment]
所以正确的做法是在渲染页面时重置 MathJax 的 label? 如果是, 我应该如何实现呢?

**Comment by Mister-Hope** [comment]
Can you provide contents which can trigger error you mentioned?

**Comment by Mister-Hope** [comment]
The problem is that VuePress provide only one markdown parser and use it to render pages parallelly. So it's impossible to use different mathjax across pages.

If using different instance per block, too much memory will be taken, there are performance issues and the label is actually not working across math blocks and tags, and it's hard to get final style.

Anyway necessary improvement and fixes are made upstream, but cannot apply in VuePress.

**Comment by Mister-Hope** [comment]
I am not sure if this pr actually fix the problem as pages are rendered by `Promise.all()`, if it still can not solve the issue, then you should avoid use thing like label, maybe you should try to place them with `\tag{}`

**Comment by liblaf** [comment]
> I am not sure if this pr actually fix the problem as pages are rendered by `Promise.all()`, if it still can not solve the issue, then you should avoid use thing like label, maybe you should try to place them with `\tag{}`

THX, I will test it later. BTW, the reason I use `\label{}` rather than `\tag{}` is that I want to refer to the equations with `\eqref{}` in inline math expressions. So `\tag{}` cannot completely replace `\label{}`, but I will try to avoid using `\label{}`.

**Comment by Mister-Hope** [comment]
So does new version fix your issue?

**Comment by liblaf** [comment]
> So does new version fix your issue?

Yes, it works. But I found a new bug regarding sidebar, which of course have nothing to do with mathjax. I will report it as a new issue later.
